sct-commons/src/main/java/org/lfenergy/compas/sct/commons/scl/PrivateService.java

```diff
@@ -239,13 +241,122 @@ private static TPrivate createPrivate(JAXBElement<?> jaxbElement) {
+                    PrivateLinkedToSTDs privateLinkedToSTDs = stringSCLMap.get(icdSysVer);
+                    List<SCL> list = privateLinkedToSTDs != null ? privateLinkedToSTDs.stdList() : new ArrayList<>();
+                    list.add(std);
+                    stringSCLMap.put(icdSysVer, new PrivateLinkedToSTDs(tp, list));
+                })
+        )));
+        return stringSCLMap;
+    }
 
 
+    public record PrivateLinkedToSTDs (TPrivate tPrivate, List<SCL> stdList) {
+    }
 
 
+    /**
+     * Checks SCD and STD compatibilities by checking if there is at least one ICD_SYSTEM_VERSION_UUID in
+     * Substation/../LNode/Private COMPAS-ICDHeader of SCL not present in IED/Private COMPAS-ICDHeader of STD
+     *
+     * @param mapICDSystemVersionUuidAndSTDFile map of ICD_SYSTEM_VERSION_UUID and list of corresponding STD
+     * @throws ScdException throws when there are several STD files corresponding to <em>ICD_SYSTEM_VERSION_UUID</em>
+     *                      from Substation/../LNode/Private COMPAS-ICDHeader of SCL
+     */
+    public static void checkSTDCorrespondanceWithLNodeCompasICDHeader(Map<String, PrivateLinkedToSTDs> mapICDSystemVersionUuidAndSTDFile) throws ScdException {
+        for (PrivateLinkedToSTDs privateLinkedToSTDs : mapICDSystemVersionUuidAndSTDFile.values()) {
+            if (privateLinkedToSTDs.stdList().size() != 1) {
+                TPrivate key = privateLinkedToSTDs.tPrivate();
+                throw new ScdException("There are several STD files corresponding to " + stdCheckFormatExceptionMessage(key));
+            }
+        }
+    }
 
+    /**
+     * Creates formatted message including data's of Private for Exception
+     *
+     * @param key Private causing exception
+     * @return formatted message
+     * @throws ScdException throws when parameter not present in Private
+     */
+    public static String stdCheckFormatExceptionMessage(TPrivate key) throws ScdException {
+        Optional<TCompasICDHeader> optionalCompasICDHeader = PrivateService.extractCompasICDHeader(key);
+        return  HEADER_ID + " = " + optionalCompasICDHeader.map(TCompasICDHeader::getHeaderId).orElse(null) + " " +
+                HEADER_VERSION + " = " + optionalCompasICDHeader.map(TCompasICDHeader::getHeaderVersion).orElse(null) + " " +
+                HEADER_REVISION + " = " + optionalCompasICDHeader.map(TCompasICDHeader::getHeaderRevision).orElse(null) +
+                " and " + ICD_SYSTEM_VERSION_UUID + " = " + optionalCompasICDHeader.map(TCompasICDHeader::getICDSystemVersionUUID).orElse(null);
+    }
 
 
+    /**
+     * Creates map of IEDName and related Private for all Privates COMPAS-ICDHeader in /Substation of SCL
+     *
+     * @param scdRootAdapter SCL file in which Private should be found
+     * @return map of Private and its IEDName parameter
+     */
+    public static Map<String, TPrivate> createMapIEDNameAndPrivate(SclRootAdapter scdRootAdapter) {
```

Comment:
this method is only used to populate the `scdRootAdapter` so we could return a stream of `Stream<TPrivate>` and use a `foreach` on this stream

Reply:
ok